src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -127,113 +131,67 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
-            FilterValueDefinition.String(String.valueOf(value))))
-      case EqualTo(attribute, value) if attribute.equalsIgnoreCase("externalId") =>
-        Right(
-          FilterDefinition.Equals(
-            createNodeOrEdgeCommonAttributeRef(instanceType, "externalId"),
+            createNodeOrEdgeCommonAttributeRef(instanceType, attribute),
             FilterValueDefinition.String(String.valueOf(value))))
       case EqualTo(attribute, value: GenericRowWithSchema)
-          if attribute.equalsIgnoreCase("startNode") && instanceType == InstanceType.Edge =>
-        createEqualsAttributeFilter("startNode", value, instanceType)
-      case EqualTo(attribute, value: GenericRowWithSchema)
-          if attribute.equalsIgnoreCase("endNode") && instanceType == InstanceType.Edge =>
-        createEqualsAttributeFilter("endNode", value, instanceType)
-      case EqualTo(attribute, value: GenericRowWithSchema) if attribute.equalsIgnoreCase("_type") =>
-        createEqualsAttributeFilter("type", value, instanceType)
-      //TODO add exists on view property first
-      case EqualTo(attribute, value: GenericRowWithSchema)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        createEqualsAttributeFilter("type", value, instanceType)
-      case Or(f1, f2) =>
-        Vector(f1, f2)
-          .traverse(toNodeOrEdgeAttributeFilter(instanceType, _))
-          .map(FilterDefinition.Or.apply)
-      case And(f1, f2) =>
-        Vector(f1, f2)
-          .traverse(toNodeOrEdgeAttributeFilter(instanceType, _))
-          .map(FilterDefinition.And.apply)
-      case Not(f) =>
-        toNodeOrEdgeAttributeFilter(instanceType, f)
-          .map(FilterDefinition.Not.apply)
-      //Node types are optional so exists will be automatically checked
-      //Types are a property of the node/edge and should not use the view but directly check on the node
-      case IsNotNull(attribute)
-          if attribute.equalsIgnoreCase("_type") && instanceType == InstanceType.Node =>
-        Right(FilterDefinition.Exists(Seq("node", "type")))
-      case IsNotNull(attribute)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        Right(FilterDefinition.Exists(Seq("edge", "type")))
-      case IsNull(attribute)
-          if attribute.equalsIgnoreCase("_type") && instanceType == InstanceType.Node =>
-        Right(FilterDefinition.Not(FilterDefinition.Exists(Seq("node", "type"))))
-      case IsNull(attribute)
-          if attribute.equalsIgnoreCase("type") && instanceType == InstanceType.Edge =>
-        Right(FilterDefinition.Not(FilterDefinition.Exists(Seq("edge", "type"))))
+          if nodeOrEdgeReferenceAttributes.contains(attribute) =>
+        createEqualsAttributeFilter(createNodeOrEdgeCommonAttributeRef(instanceType, attribute), value)
+      case In(attribute, values) if nodeOrEdgeStringAttributes.contains(attribute) =>
```

Comment:
Looks like In&StringStartsWith handling wasn't here before, did it move in because of refactoring or it's adding/fixing support for those cases?

Reply:
It was in the other filter not this one, so yeah this is adding/fixing. Presumably in case of sync, only these filters were called so this adds support for these pushdown filters in the sync endpoints, I think in other places the two filters were chained.

Reply:
For sync actually there was effort to remove most of pushdown as it may be not backend by indexes :eyes:

Reply:
Though... I think sync filters had filtering on these "special" attributes already (externalId, space, type, start & endNode) so there is an index on them, and that shouldn't be different if you do indexedColumn = something or indexedColumn in/startswith something... unless the implementation on the side of fdm is weird.

Reply:
Should I ask them then?

Reply:
Makes sense, if they were pushed down then equality filters should be fine, prefix a bit less certain. Won't hurt to double-check still :+1:

Reply:
no reply from fdm, what do I do there ^^?